### PR TITLE
bento4: fix on darwin, remove broken on aarch64

### PR DESCRIPTION
**Pull request by wlcx**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/video/bento4/default.nix [hunk 1]
```diff
@@ -23,17 +23,21 @@ stdenv.mkDerivation rec {
   installPhase = ''
     runHook preInstall
     mkdir -p $out/{lib,bin}
-    find -iname '*.so' -exec mv --target-directory="$out/lib" {} \;
+    find -iname '*.so' -o -iname '*.dylib' -exec mv --target-directory="$out/lib" {} \;
```
Comment (wegank) [inline pkgs/tools/video/bento4/default.nix:26]:
```suggestion
    find -iname '*${stdenv.hostPlatform.extensions.sharedLibrary}' -exec mv --target-directory="$out/lib" {} \;
```

Comment (wlcx) [reply pkgs/tools/video/bento4/default.nix:26]: Amended, thanks!

File: pkgs/tools/video/bento4/default.nix [hunk 2]
```diff
@@ -23,17 +23,21 @@ stdenv.mkDerivation rec {
...
     runHook preInstall
     mkdir -p $out/{lib,bin}
-    find -iname '*.so' -exec mv --target-directory="$out/lib" {} \;
+    find -iname '*${stdenv.hostPlatform.extensions.sharedLibrary}' -exec mv --target-directory="$out/lib" {} \;
     find -maxdepth 1 -executable -type f -exec mv --target-directory="$out/bin" {} \;
     runHook postInstall
   '';
 
+  # Patch binaries to use our dylib
+  fixupPhase = lib.optional stdenv.isDarwin ''
```
Comment (wegank) [inline pkgs/tools/video/bento4/default.nix:32]:
```suggestion
  postInstall = lib.optionalString stdenv.isDarwin ''
```

Comment (wlcx) [reply pkgs/tools/video/bento4/default.nix]: Amended, thanks. Can I ask for some background on why? Very much still learning the ins and outs of nixpkgs.

**Review by makefu — APPROVED** [review]
cannot test on darwin but patch looks good

File: pkgs/tools/video/bento4/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/tools/video/bento4/default.nix]: The default phases are [not void](https://nixos.org/manual/nixpkgs/stable/#ssec-fixup-phase), and overriding `fixupPhase` with `""` on Linux is essentially doing `dontFixup = true`.

**Pull request merged**